src/uct/api/uct.h

```diff
@@ -1108,6 +1132,31 @@ ucs_status_t uct_iface_set_am_tracer(uct_iface_h iface, uct_am_tracer_t tracer,
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief Create an interface for listening on a socket address and creating
+ *        a connection to remote peer.
+ *
+ * This routine will create an interface which will listen to the given socket address.
+ * The user defined callback will be invoked upon connection @ref
+ * uct_conn_event_type_t events arriving while listening.
+ *
+ * @param [in]  worker           Handle to worker which will be associated with
+ *                               the new interface.
+ * @param [in]  md               Memory domain to create the interface on.
+ *                               This memory domain must support the @ref
+ *                               UCT_MD_FLAG_SOCK_ADDR flag.
+ * @param [in]  sockaddr_params  The sockaddr to listen on and a callback for
+ *                               incoming events.
+ * @param [in]  config           Interface configuration options.
+ * @param [out] iface_p          Handle to the created interface.
+ */
+ucs_status_t uct_sockaddr_listen(uct_worker_h worker, uct_md_h md,
```

Comment:
per the discussion on conf call: need to re-use uct_iface_open() function for this, and put the sockaddr params inside uct_iface_params_t